VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -278,85 +274,63 @@ private void updateUIForVideo(int position, View convertView, final ViewHolder v
-                                    logger.debug("Download Button Clicked");
-                                    //notifyDataSetChanged();
-                                    mDownloadListener.download(videoData);
-                                }
-                            });
-                        } else if (ds == DownloadEntry.DownloadedState.DOWNLOADING) {
-                            // may be download in progress
-                            EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
-                            viewHolder.bulkDownload.setVisibility(View.GONE);
-                            storage.getDownloadProgressByDmid(videoData.dmId, new DataCallback<Integer>(true) {
-                                @Override
-                                public void onResult(Integer result) {
-                                    if (result >= 0 && result < 100) {
-                                        EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.STARTED));
-                                    } else if (result == 100) {
-                                        EventBus.getDefault().post(new DownloadEvent(DownloadEvent.DownloadStatus.COMPLETED));
-                                    }
-                                }
-
-                                @Override
-                                public void onFail(Exception ex) {
-                                    logger.error(ex);
-                                    viewHolder.bulkDownload.setVisibility(View.VISIBLE);
-                                }
-                            });
-                        } else if (ds == DownloadEntry.DownloadedState.DOWNLOADED) {
-                            // downloaded
-                            viewHolder.bulkDownload.setVisibility(View.GONE);
+                    public void onResult(DownloadEntry.WatchedState result) {
+                        if (result != null && result == DownloadEntry.WatchedState.WATCHED) {
+                            viewHolder.rowType.setIconColorResource(R.color.edx_grayscale_neutral_base);
+                        } else {
+                            viewHolder.rowType.setIconColorResource(R.color.edx_brand_primary_base);
                         }
-
                     }
 
                     @Override
                     public void onFail(Exception ex) {
                         logger.error(ex);
-                        viewHolder.bulkDownload.setVisibility(View.VISIBLE);
                     }
                 });
 
+        if (videoData.isVideoForWebOnly()) {
+            viewHolder.numOfVideoAndDownloadArea.setVisibility(View.GONE);
+        }
+        else {
+            viewHolder.numOfVideoAndDownloadArea.setVisibility(View.VISIBLE);
+            dbStore.getDownloadedStateForVideoId(videoData.videoId,
+                    new DataCallback<DownloadEntry.DownloadedState>(true) {
```

Comment:
For now row-recycling has been disabled and following JIRA issue has been opened:
https://openedx.atlassian.net/browse/MA-1640